apps/server/src/modules/tool/uc/school-external-tool.uc.spec.ts

```diff
@@ -239,4 +239,70 @@ describe('SchoolExternalToolUc', () => {
 			});
 		});
 	});
+
+	describe('updateSchoolExternalTool is called', () => {
+		const setupUpdate = () => {
+			const { tool } = setup();
+			const updatedTool: SchoolExternalToolDO = { ...tool };
+			updatedTool.parameters[0].value = 'updatedValue';
+
+			schoolExternalToolService.saveSchoolExternalTool.mockResolvedValue(updatedTool);
+			return {
+				updatedTool,
+			};
+		};
+
+		describe('when checks permission', () => {
```

Comment:
I don't think it makes sense to have this and the below describe-block because they describe the same scenario. The `describe('updateSchoolExternalTool is called', () => ...` is sufficient IMO.

Reply:
We want to keep the cases separate

Reply:
Why?

Reply:
@dyedwiper In this case we follow the test structure which was told us. I already asked for a brown bag sessions to present this structure because actually it depends who you ask

Reply:
Openly spoken I'm not a fan of this test structure. I don't mean to take it into question here though. I just think you're not using it correctly. As far as I got the structure, each scenario should have its own describe-block, opposed to having the differentiation of scenarios at the it-level.

An example:
`describe('when user is not authorized', () => { const setup = ...; it('should throw ForbiddenException', () => ...`
opposed to
`it('should throw ForbiddenException if user is not authorized', () => ...`

But you don't have different scenarios here. All four tests test the normal flow of the method and have the same setup.

Reply:
I know what you mean but i cant get what is the advantage to have one describe block when you only have one test in it... but i think we need a session for that, maybe i can get it then.
If you insist on the changes we will of course do it :)

Reply:
I agree with you. That's part of my criticism on the structure. 

But like I said: Even in this structure these describe-blocks here were unnecessary.